app/Platform/Actions/BuildGameListAction.php

```diff
@@ -181,4 +172,105 @@ private function getUnfilteredResultsCount(Builder $query, array $filters): ?int
+            }
+
+            $sourceGame = Game::findOrFail($targetId);
+
+            return (new GameSuggestionEngine($user, $sourceGame))->selectSuggestions();
+        }
+
+        return [];
+    }
+
+    /**
+     * Check if the given list type is a suggestion-based list.
+     */
+    private function isSuggestionListType(GameListType $listType): bool
+    {
+        return in_array($listType, [
+            GameListType::UserSpecificSuggestions,
+            GameListType::GameSpecificSuggestions,
+        ]);
+    }
+
+    /**
+     * Transform game list entries into their corresponding DTO types.
+     * For suggestion lists, this returns GameSuggestionEntryData objects.
+     * For regular lists, this returns GameListEntryData objects.
+     *
+     * @param Collection<int, Game> $entries
+     * @param Collection<int, PlayerGame> $playerGames
+     * @param Collection<int|string, int|string> $backlogGames
+     * @return array<GameListEntryData|GameSuggestionEntryData>
+     */
+    private function transformGameListEntries(
+        Collection $entries,
+        GameListType $listType,
+        ?User $user,
+        Collection $playerGames,
+        Collection $backlogGames
+    ): array {
+        return $entries->map(function (Game $game) use ($listType, $user, $playerGames, $backlogGames): GameListEntryData {
+            $playerGame = $playerGames->get($game->id);
+
+            $baseData = [
+                'game' => GameData::from($game)->include(
+                    'achievementsPublished',
+                    'badgeUrl',
+                    'hasActiveOrInReviewClaims',
+                    'lastUpdated',
+                    'numVisibleLeaderboards',
+                    'playersTotal',
+                    'pointsTotal',
+                    'pointsWeighted',
+                    'releasedAt',
+                    'releasedAtGranularity',
+                    'system.iconUrl',
+                    'system.nameShort',
+                    $user?->can('develop') ? 'numUnresolvedTickets' : '',
+                ),
+                'playerGame' => $playerGame
+                    ? PlayerGameData::fromPlayerGame($playerGame)->include('highestAward')
+                    : null,
+                'isInBacklog' => $listType === GameListType::UserPlay
+                    ? true
+                    : $backlogGames->has($game->id),
+            ];
+
+            // Game suggestions have contextual data attached about why
+            // that particular game was suggested to the user.
+            if ($this->isSuggestionListType($listType)) {
+                $suggestionData = collect($this->suggestions)
+                    ->first(fn ($suggestion) => $suggestion->gameId === $game->id);
+
+                $suggestionEntryData = $baseData + [
+                    'suggestionReason' => $suggestionData->reason,
+                    'relatedGameContext' => $suggestionData->context,
```

Comment:
I understand the desire for this on the full game list, but when there's a source game, should the source game be duplicated for every similar game entry?
![image](https://github.com/user-attachments/assets/f4b97e56-0ec9-4533-95a2-66128354a426)